expression/expr_to_pb.go

```diff
@@ -198,124 +198,34 @@ func (pc PbConverter) columnToPBExpr(column *Column) *tipb.Expr {
 }
 
 func (pc PbConverter) scalarFuncToPBExpr(expr *ScalarFunction) *tipb.Expr {
-	switch expr.FuncName.L {
-	case ast.LT, ast.LE, ast.EQ, ast.NE, ast.GE, ast.GT,
-		ast.NullEQ:
-		return pc.compareOpsToPBExpr(expr)
-	case ast.Like:
-		return pc.likeToPBExpr(expr)
-	case ast.Plus, ast.Minus, ast.Mul, ast.Div:
-		return pc.arithmeticalOpsToPBExpr(expr)
-	case ast.LogicAnd, ast.LogicOr, ast.UnaryNot, ast.LogicXor:
-		return pc.logicalOpsToPBExpr(expr)
-	case ast.And, ast.Or, ast.BitNeg, ast.Xor:
-		return pc.bitwiseFuncToPBExpr(expr)
-	case ast.Case, ast.Coalesce, ast.If, ast.Ifnull, ast.IsNull, ast.IsTruth, ast.IsFalsity, ast.In:
-		return pc.builtinFuncToPBExpr(expr)
-	case ast.JSONType, ast.JSONExtract, ast.JSONUnquote, ast.JSONValid,
-		ast.JSONObject, ast.JSONArray, ast.JSONMerge, ast.JSONSet,
-		ast.JSONInsert, ast.JSONReplace, ast.JSONRemove, ast.JSONContains:
-		return pc.jsonFuncToPBExpr(expr)
-	case ast.DateFormat:
-		return pc.dateFuncToPBExpr(expr)
-	default:
+	// check whether this function can be pushed.
+	if !pc.canFuncBePushed(expr) {
 		return nil
 	}
-}
-
-func (pc PbConverter) compareOpsToPBExpr(expr *ScalarFunction) *tipb.Expr {
-	var tp tipb.ExprType
-	switch expr.FuncName.L {
-	case ast.LT:
-		tp = tipb.ExprType_LT
-	case ast.LE:
-		tp = tipb.ExprType_LE
-	case ast.EQ:
-		tp = tipb.ExprType_EQ
-	case ast.NE:
-		tp = tipb.ExprType_NE
-	case ast.GE:
-		tp = tipb.ExprType_GE
-	case ast.GT:
-		tp = tipb.ExprType_GT
-	case ast.NullEQ:
-		tp = tipb.ExprType_NullEQ
-	}
-	return pc.convertToPBExpr(expr, tp)
-}
 
-func (pc PbConverter) likeToPBExpr(expr *ScalarFunction) *tipb.Expr {
-	if !pc.client.IsRequestTypeSupported(kv.ReqTypeSelect, int64(tipb.ExprType_Like)) {
```

Comment:
Why remove this check?

Reply:
in the old code logic, it was checked in `convertToPBExpr`, in the new code logic, it is checked in `canFuncBePushed`